File: android/src/main/java/com/androidphotopicker/Options.java
package com.androidphotopicker;

import com.facebook.react.bridge.ReadableMap;

import android.text.TextUtils;

public class Options {
  int selectionLimit = 1;
  int videoQuality = 1;
  int quality;
  int conversionQuality = 92;
  Boolean convertToJpeg = true;
  int maxWidth;
  int maxHeight;
  Boolean useFrontCamera = false;
  String mediaType;


  Options(ReadableMap options) {
    mediaType = options.getString("mediaType");

    if(options.hasKey("selectionLimit")) {
      selectionLimit = options.getInt("selectionLimit");
    }

    String videoQualityString = options.getString("videoQuality");
    if (!TextUtils.isEmpty(videoQualityString) && !videoQualityString.toLowerCase().equals("high")) {
      videoQuality = 0;
    }

    if (options.hasKey("conversionQuality")) {
      conversionQuality = (int) (options.getDouble("conversionQuality") * 100);
    }

    String assetRepresentationMode = options.getString("assetRepresentationMode");
    if (!TextUtils.isEmpty(assetRepresentationMode) && assetRepresentationMode.toLowerCase().equals("current")) {
      convertToJpeg = false;
    }

    if (options.getString("cameraType").equals("front")) {
      useFrontCamera = true;
    }

    quality = (int) (options.getDouble("quality") * 100);

    if (options.hasKey("maxHeight")) {
      maxHeight = options.getInt("maxHeight");
    }

    if (options.hasKey("maxWidth")) {
      maxWidth = options.getInt("maxWidth");
    }
  }
}

File: android/src/main/java/com/androidphotopicker/Options.kt
package com.androidphotopicker

import com.facebook.react.bridge.ReadableMap

class Options(options: ReadableMap) {
    var selectionLimit: Int = 1
    var videoQuality: Int = 1
    var quality: Int
    var conversionQuality: Int = 92
    var convertToJpeg: Boolean = true
    var maxWidth: Int = 0
    var maxHeight: Int = 0
    var useFrontCamera: Boolean = false
    var mediaType: String? = options.getString("mediaType")

    init {
        if (options.hasKey("selectionLimit")) {
            selectionLimit = options.getInt("selectionLimit")
        }

        val videoQualityString = options.getString("videoQuality")
        if (!videoQualityString.isNullOrEmpty() && videoQualityString.lowercase() != "high") {
            videoQuality = 0
        }

        if (options.hasKey("conversionQuality")) {
            conversionQuality = (options.getDouble("conversionQuality") * 100).toInt()
        }

        val assetRepresentationMode = options.getString("assetRepresentationMode")
        if (!assetRepresentationMode.isNullOrEmpty() && assetRepresentationMode.lowercase() == "current") {
            convertToJpeg = false
        }

        if (options.getString("cameraType") == "front") {
            useFrontCamera = true
        }

        quality = (options.getDouble("quality") * 100).toInt()

        if (options.hasKey("maxHeight")) {
            maxHeight = options.getInt("maxHeight")
        }

        if (options.hasKey("maxWidth")) {
            maxWidth = options.getInt("maxWidth")
        }
    }
}

